.github/workflows/dependencies_image.yml

```diff
@@ -0,0 +1,47 @@
+name: 'OpenTelemetry-cpp dependencies image'
+on:
+  schedule:
+    - cron: "0 3 * * 6"
+
+jobs:
+  docker_image:
+    name: Docker Image
+    runs-on: ubuntu-latest
+    timeout-minutes: 300
+    steps:
+    -
+      name: checkout
+      uses: actions/checkout@v2
+    -
+      name: Set up QEMU
+      uses: docker/setup-qemu-action@v1
+    -
+      name: Set up Docker Buildx
+      id: buildx
+      uses: docker/setup-buildx-action@v1
+    -
+      name: Build Image
+      uses: docker/build-push-action@v2
+      with:
+        builder: ${{ steps.buildx.outputs.name }}
+        context: ci/
+        file: ./docker/Dockerfile
+        build-args: BASE_IMAGE=ubuntu:latest
+        platforms: linux/amd64
+        # platforms: linux/amd64,linux/arm64
+        push: false
+        tags: otel-cpp
+        load: true
+        cache-from: type=gha
+        # cache-to: type=gha,mode=max
+    -
+      name: Save Image
+      run: |
+        docker images
+        docker save -o /opt/otel-cpp-deps.tar otel-cpp
+    -
+      name: Upload Image
+      uses: actions/upload-artifact@v3
+      with:
+        name: otel-cpp-deps
+        path: /opt/otel-cpp-deps.tar
```

Comment:
Do we have the estimated size of these tar-ed docker images, as GitHub cache is limited to 10GB, and we are already using some of it for Bazel builds? We can later also see if these artifacts can be stored separately in azure blob - 
https://github.com/marketplace/actions/upload-a-build-artifact-to-azure-blob-storage

Reply:
I didn't find the size limit in the [documentation](https://github.com/actions/upload-artifact/issues/9). I think the same rule as logs should apply here. I'm going to reduce the retention period  for the artifact to two weeks.

The size of the artifact is about [1GB](https://github.com/esigo/opentelemetry-cpp/actions/runs/2181104518).

We can reduce this size (probably to 200MB) by storing `gRPC` and `thrift` only in the image.

Reply:
Ok interesting. I also don't see it is documented too. I was thinking it to be part of Cache size limit.
What all we store right now apart from gRPC and thrift ?

Reply:
the image had everything for building the library git, cmake,...
I changed the docker file. now it has only `gRPC` and `thrift`.